plugins/rating/services/RatingService.php

```diff
@@ -0,0 +1,156 @@
+	 * @return int
+	 */
+	public function rateAction ($entryId, $rank)
+	{
+		if (!$entryId)
+		{
+			throw new KalturaAPIException(KalturaErrors::MISSING_MANDATORY_PARAMETER, "entryId");
+		}
+		
+		$dbEntry = entryPeer::retrieveByPK($entryId);
+		if (!$dbEntry)
+		{
+			throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $entryId);
+		}
+		
+		//Check if a kvote for current entryId and kuser already exists.
+		$existingKVote = kvotePeer::doSelectByEntryIdAndPuserId($entryId, $this->getPartnerId(), kCurrentContext::$ks_uid);
+		if ($existingKVote)
+		{
+			$existingKVote->setRank($rank);
+			$existingKVote->save();
+		}
+		else
+		{
+			kvotePeer::createKvote($entryId, $this->getPartnerId(), kCurrentContext::$ks_uid, $rank, KVoteType::RANK);
+		}
+		
+		return $rank;
+	}
+	
+	/**
+	 * @action removeRating
+	 * Action to remove current kuser's rating for a specific entry
+	 * @param string $entryId
+	 * @throws KalturaErrors::ENTRY_ID_NOT_FOUND
+	 * @throws KalturaRatingErrors::USER_RATING_FOR_ENTRY_NOT_FOUND
+	 * @return bool
+	 */
+	public function removeRatingAction ($entryId)
+	{
+		if (!$entryId)
+		{
+			throw new KalturaAPIException(KalturaErrors::MISSING_MANDATORY_PARAMETER, "entryId");
+		}
+		
+		$dbEntry = entryPeer::retrieveByPK($entryId);
+		if (!$dbEntry)
+		{
+			throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $entryId);
+		}
+		
+		$existingKVote = kvotePeer::doSelectByEntryIdAndPuserId($entryId, $this->getPartnerId(), kCurrentContext::$ks_uid);
+		if (!$existingKVote)
+		{
+			throw new KalturaAPIException(KalturaRatingErrors::USER_RATING_FOR_ENTRY_NOT_FOUND);
+		}
+		
+		
+		if (kvotePeer::disableExistingKVote($entryId, $this->getPartnerId(), kCurrentContext::$ks_uid))
+			return true;
+		
+		return false;
+	}
+	
+	/**
+	 * @action checkRating
+	 * Action to check current kuser's rating for a specific entry
+	 * @param string $entryId
+	 * @throws KalturaErrors::ENTRY_ID_NOT_FOUND
+	 * @return int
+	 */
+	public function checkRatingAction ($entryId)
+	{
+		if (!$entryId)
+		{
+			throw new KalturaAPIException(KalturaErrors::MISSING_MANDATORY_PARAMETER, "entryId");
+		}
+		
+		$dbEntry = entryPeer::retrieveByPK($entryId);
+		if (!$dbEntry)
+		{
+			throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $entryId);
+		}
+		
+		$existingKVote = kvotePeer::doSelectByEntryIdAndPuserId($entryId, $this->getPartnerId(), kCurrentContext::$ks_uid);
+		if (!$existingKVote)
+		{
+			throw new KalturaAPIException(KalturaRatingErrors::USER_RATING_FOR_ENTRY_NOT_FOUND);
+		}
+		
+		return $existingKVote->getRank();
+		
+	}
+	
+	
+	/**
+	 * @action getRatingCounts
+	 * Action to check entry's rating counts breakdown
+	 * @param KalturaRatingCountFilter $filter
+	 * @throws KalturaErrors::ENTRY_ID_NOT_FOUND
+	 * @throws KalturaRatingErrors::USER_RATING_FOR_ENTRY_NOT_FOUND
+	 * @return KalturaRatingCountListResponse
+	 */
+	public function getRatingCountsAction (KalturaRatingCountFilter $filter)
```

Comment:
If you want to allow calling this function with empty filter than you need to add default value null to the function signature
But, looking at getListResponse of this filter - you must have a valid filter with entry ID set , so this option is impossible